python/pyarrow/array.pxi

```diff
@@ -2115,6 +2116,22 @@ cdef dict _array_classes = {
 }
 
 
+cdef object get_array_class_from_type(
+        const shared_ptr[CDataType]& sp_data_type):
+    cdef CDataType* data_type = sp_data_type.get()
+    if data_type == NULL:
+        raise ValueError('Array data type was NULL')
+
+    if data_type.id() == _Type_EXTENSION:
+        py_ext_data_type = pyarrow_wrap_data_type(sp_data_type)
+        py_ext_array_class = py_ext_data_type.__arrow_ext_class__()
+        if py_ext_array_class is NotImplementedError:
+            py_ext_array_class = ExtensionArray
```

Comment:
Alternatively, that method could also return ExtensionArray by default?

Reply:
Make sense. I followed the existing convention, but the logic is cleaner if ExtensionArray is the default value. Now updated.